### ISSUE TITLE
SpeedDial: Tabbing issues

### ISSUE THREAD

**Issue by Teutron-flo** [primefaces/primefaces#14234]
### Describe the bug

Right now you can not select with TAB menu items inside a SpeedDial.
 
This can be reproduced on the showcase page:
https://showcase.primefaces.org/ui/button/speedDial.xhtml

The reason for the issue is that the elements are rendered with tabindex -1:

    <a tabindex="-1" role="menuitem" class="ui-speeddial-action" href="#" 
        onclick="PrimeFaces.ab({s:&quot;speedDialForm:j_id_4j&quot;,f:&quot;speedDialForm
        &quot;,u:&quot;speedDialForm:messages    &quot;});return false;">
    <span class="ui-speeddial-action-icon pi pi-refresh" aria-hidden="true"></span>
    <span class="ui-menuitem-text"></span>
    </a>

And the value will not be changed on show / hide.

For my point of view, with a simple change inside primefaces/src/main/frontend/packages/components/src/speeddial/speeddial.widget.js file
we can activate the tabbing also for the menu items:

    show: function() {
        this.mask && this.mask.addClass("ui-speeddial-mask-visible");
        this.container.addClass("ui-speeddial-opened");

        // new Code
        this.container.find('a').attr("tabindex", "0"); 
        
        this.visible = !0;
        this.updateItemStyles();
        this.cfg.onVisibleChange && this.cfg.onVisibleChange.call(this, !0);
        this.cfg.onShow && this.cfg.onShow.call(this)
    },
    hide: function() {
        this.mask && this.mask.removeClass("ui-speeddial-mask-visible");
        this.container.removeClass("ui-speeddial-opened");

         // new Code
        this.container.find('a').attr("tabindex", "-1");
        
        this.visible = !1;
        this.updateItemStyles();
        this.cfg.onVisibleChange && this.cfg.onVisibleChange.call(this, !1);
        this.cfg.onHide && this.cfg.onHide.call(this)
    },

Also for the main button and the subelements is a css class missing to mark the object with a box shadow incase it has focus:

	.ui-speeddial-button:focus {
		outline: 0 none;
		outline-offset: 0;
		box-shadow: var(--focus-ring);
	}

	.ui-speeddial-action:focus {
		outline: 0 none;
		outline-offset: 0;
		box-shadow: var(--focus-ring);
	}

### PrimeFaces edition

None

### PrimeFaces version

15.0.9

### JSF implementation

All

### JSF version

4.1.0

### Java version

21

**Comment by melloware** [comment]
Can you submit a PR?

**Comment by Teutron-Florian** [comment]
[14234_test.zip](https://github.com/user-attachments/files/22737582/14234_test.zip)

Project to test the changes

**Comment by Teutron-Florian** [comment]
@melloware PR is in place also a test project
